django/views/templates/default_urlconf.html

```diff
@@ -1,19 +1,16 @@
 {% load i18n %}
 <!doctype html>
 {% get_current_language_bidi as LANGUAGE_BIDI %}
-<html{% if LANGUAGE_BIDI %} dir="rtl"{% endif %}>
+<html lang="{{ LANGUAGE_CODE|default:"en-us" }}"{% if LANGUAGE_BIDI %} dir="rtl"{% endif %}>
```

Comment:
The value in the [`lang`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/lang) attribute should comply with [BCP47](https://www.ietf.org/rfc/bcp/bcp47.txt). It does allow for the value to be specified case-insensitively, but prefers that the region subtag is uppercase and the script subtag is titlecase. In other words, the default value `en-us` ought to be `en-US` and the value of `LANGUAGE_CODE` would need to be converted. In practice, however, we will just have to ignore this as it is a pain to convert and should still work.

It's probably a good idea to be explicit with the [`dir`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/dir) attribute?

```suggestion
<html lang="{{ LANGUAGE_CODE|default:'en-us' }}" dir="{{ LANGUAGE_BIDI|yesno:'rtl,ltr,auto' }}">
```

Reply:
Hmm, this really isn’t my strong point. I’m happy to apply your suggested change, but it’s not clear to me what issue this addresses if any?

Note the code I used here is copied straight from Django’s base template (apart from me forgetting `get_current_language`), so if we do set `lang` or `dir` differently here, it probably makes sense to update it there (and in all other templates, but that’s for another day).

https://github.com/django/django/blob/11c4a4412b74bb1dfe52d706a58f230066821c33/django/contrib/admin/templates/admin/base.html#L3

Reply:
I moved it to a separate commit.